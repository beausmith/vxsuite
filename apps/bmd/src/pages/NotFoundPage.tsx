import React from 'react'
import { RouteComponentProps } from 'react-router-dom'

import LinkButton from '../components/LinkButton'
import Main, { MainChild } from '../components/Main'
import Prose from '../components/Prose'

const NotFoundPage = (props: RouteComponentProps) => {
  const { pathname } = props.location
  return (
    <Main>
      <MainChild center>
        <Prose textCenter>
          <h1>Page Not Found.</h1>
          <p>
            No page exists at <code>{pathname}</code>.
          </p>
          <p>
            <LinkButton primary to={`/`}>
              Start Page
            </LinkButton>
          </p>
        </Prose>
      </MainChild>
    </Main>
  )
}

export default NotFoundPage
