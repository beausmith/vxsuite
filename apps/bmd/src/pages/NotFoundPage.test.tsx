import React from 'react'
import { Route } from 'react-router-dom'
import { fireEvent } from 'react-testing-library'

import { render } from '../../test/testUtils'

import NotFoundPage from './NotFoundPage'

it(`renders NotFoundPage`, () => {
  const resetBallot = jest.fn()
  const { container, getByText } = render(
    <Route path="/" component={NotFoundPage} />,
    {
      resetBallot,
      route: '/foobar-not-found-path',
    }
  )
  expect(container.firstChild).toMatchSnapshot()
  fireEvent.click(getByText('Start Over'))
  expect(resetBallot).toHaveBeenCalled()
})
