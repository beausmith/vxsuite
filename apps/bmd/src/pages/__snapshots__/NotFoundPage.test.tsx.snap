// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotFoundPage 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
  padding: 1rem 0.5rem 2rem;
}

.c1 {
  max-width: 35rem;
  margin: auto auto;
}

.c2 {
  line-height: 1.2;
  max-width: 66ch;
  text-align: center;
  margin: auto;
}

.c2 h1 {
  font-size: 1.5rem;
  line-height: 1.1;
  margin: 2rem 0 1rem;
}

.c2 h2 {
  font-size: 1.25rem;
  margin: 1.5rem 0 0.75rem;
}

.c2 h3,
.c2 p {
  font-size: 1rem;
  margin: 1rem 0;
}

.c2 h1 + h2 {
  margin-top: -0.75rem;
}

.c2 h1 + p,
.c2 h2 + p {
  margin-top: -0.75rem;
}

.c2 h3 + p {
  margin-top: -1rem;
}

.c2 :first-child {
  margin-top: 0;
}

.c2 :last-child {
  margin-bottom: 0;
}

.c3 {
  box-sizing: border-box;
  cursor: pointer;
  background: lightgrey;
  border: none;
  border-radius: 0.25rem;
  padding: 0.4rem 0.7rem;
  color: black;
  line-height: 1;
  white-space: nowrap;
}

@media (min-width:480px) {
  .c0 {
    padding: 2rem 1rem;
  }
}

@media print {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    padding: 0;
  }
}

@media print {
  .c1 {
    max-width: 100%;
    margin: 0;
  }
}

@media (min-width:480px) {
  .c2 {
    line-height: 1.3;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1>
        Page Not Found.
      </h1>
      <p>
        No page exists at 
        <code>
          /foobar-not-found-path
        </code>
        .
      </p>
      <p>
        <button
          class="c3"
          type="button"
        >
          Start Over
        </button>
      </p>
    </div>
  </div>
</main>
`;
